activation/activation.go

```diff
@@ -301,6 +322,43 @@ func (b *Builder) generateProof() error {
 	return nil
 }
 
+func (b *Builder) waitForFirstATX(ctx context.Context) {
+	currEpoch := b.layerClock.GetCurrentLayer().GetEpoch()
+	if currEpoch == 0 { // genesis miner
+		return
+	}
+	if prev, err := b.cdb.GetPrevAtx(b.nodeID); err == nil {
+		if prev.PubLayerID.GetEpoch() == currEpoch-1 {
+			// miner has ATX in previous epoch
+			return
+		}
+	}
+
+	// first atx for this node
+	currentLayer := b.layerClock.GetCurrentLayer()
+	waitTill := (currentLayer.GetEpoch() + 1).FirstLayer()
```

Comment:
waiting one more epoch is unnecessary if this node is still before phase shift - cycle gap + broadcast

it doesn't make a difference in systests because epoch is 60s, but in practice will be inconvenient

Reply:
done.